### ISSUE TITLE
Preference pairs for PPM training

### ISSUE THREAD

**Issue by jianghc93** [microsoft/rStar#32]
Thanks for open-sourcing your work! 

In the paper, you mentioned that "For intermediate steps (except the final answer step), the positive and negative pairs share the same preceding steps." 

May I know how many of such intermediate step pairs you managed to collect for training the PPM? When using a high exploration term (e.g., c=2), it is less likely that a node will be visited multiple times. Therefore, nodes with the same preceding steps would be rare? 

Also given the limited rollouts, I believe that majority of the intermediate step pairs you collect would be for the earlier steps? (e.g., Step 1, Step 2)

**Comment by Gxy-2001** [comment]
Thanks for your attention and thoughtful question — it's a great point and something we've been focusing on closely.  

First, we analyzed the PPM training data and found that 65.7% of samples contain three or more steps, so the majority do include multi-step reasoning.  

Second, regarding MCTS, our implementation is not a standard one. For reference, please see this link:  
https://github.com/microsoft/rStar/blob/7d1139dd5b0ad8766343b5de60596eccfd429cc9/rstar_deepthink/nodes/mcts_node.py#L50  
At the beginning, each node's `u_value` is initialized to 0, so nodes with higher `q_value` are explored first rather than guaranteeing uniform exploration. This design is based on the assumption that the reward model can effectively guide the policy model. As a result, nodes preferred by the RM are explored more, leading to higher-quality PPM training data.
